indico/modules/events/editing/client/js/components/FileManager/reducer.js

```diff
@@ -0,0 +1,110 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import _ from 'lodash';
+import {combineReducers} from 'redux';
+import * as actions from './actions';
+
+export default combineReducers({
+  fileTypes: (state = [], action) => {
+    const fileType = _.find(state, {id: action.fileTypeId});
+    switch (action.type) {
+      case actions.MARK_UPLOADED:
+        action.file.state = 'added';
+        if (fileType.multiple) {
+          fileType.files = fileType.files.concat(action.file);
+        } else {
+          fileType.files = [action.file];
+        }
+        return [...state];
+
+      case actions.MARK_DELETED: {
+        const file = _.find(fileType.files, {uuid: action.fileId});
```

Comment:
```suggestion
        const file = fileType.files.find(file => file.uuid === action.fileId);
```

Reply:
Should we use `Array.prototype.find` if we're using lodash anyway and `_.find` is nicer (at leas in my opinion)?

Reply:
How is it nicer? :p

Reply:
Oh, I just realized it searches by UUID. I agree with @pferreir that it's nicer than `[].find()`...